.github/workflows/libcxx-build-and-test.yaml

```diff
@@ -199,3 +199,44 @@ jobs:
             **/CMakeError.log
             **/CMakeOutput.log
             **/crash_diagnostics/*
+  windows:
```

Comment:
When we're ready to merge this, we should add `needs: [ stage1, stage2, stage3 ]` to only trigger the Windows CI if the other stages are happy.